libraries/DataFlash/DataFlash_MAVLinkLogTransfer.cpp

```diff
@@ -155,13 +167,17 @@ void DataFlash_Class::handle_log_request_end(GCS_MAVLINK &link, mavlink_message_
     mavlink_msg_log_request_end_decode(msg, &packet);
     _in_log_download = false;
     _log_sending = false;
+    _log_sending_chan = -1;
 }
 
 /**
    trigger sending of log messages if there are some pending
  */
 void DataFlash_Class::handle_log_send(GCS_MAVLINK &link)
 {
+    if (_log_sending_chan != link.get_chan()) {
+        return;
```

Comment:
This is actually the crux of the bug here.

handle_log_send is called on every link as part of the data stream send: https://github.com/ardupilot/ardupilot/blob/master/APMrover2/GCS_Mavlink.cpp#L599

If we were to put a message here every link apart from the sending link would get the message at 50Hz :-)

Without these patches every link gets a share of the file, which isn't particularly helpful.